plugins/inputs/mcrouter/mcrouter.go

```diff
@@ -0,0 +1,255 @@
+	"ps_vsize":                                   typeInt,
+	"ps_rss":                                     typeInt,
+	"fibers_allocated":                           typeInt,
+	"fibers_pool_size":                           typeInt,
+	"fibers_stack_high_watermark":                typeInt,
+	"successful_client_connections":              typeInt,
+	"duration_us":                                typeInt,
+	"destination_max_pending_reqs":               typeInt,
+	"destination_max_inflight_reqs":              typeInt,
+	"retrans_per_kbyte_max":                      typeInt,
+	"cmd_get_count":                              typeInt,
+	"cmd_delete_out":                             typeInt,
+	"cmd_lease_get":                              typeInt,
+	"cmd_set":                                    typeInt,
+	"cmd_get_out_all":                            typeInt,
+	"cmd_get_out":                                typeInt,
+	"cmd_lease_set_count":                        typeInt,
+	"cmd_other_out_all":                          typeInt,
+	"cmd_lease_get_out":                          typeInt,
+	"cmd_set_count":                              typeInt,
+	"cmd_lease_set_out":                          typeInt,
+	"cmd_delete_count":                           typeInt,
+	"cmd_other":                                  typeInt,
+	"cmd_delete":                                 typeInt,
+	"cmd_get":                                    typeInt,
+	"cmd_lease_set":                              typeInt,
+	"cmd_set_out":                                typeInt,
+	"cmd_lease_get_count":                        typeInt,
+	"cmd_other_out":                              typeInt,
+	"cmd_lease_get_out_all":                      typeInt,
+	"cmd_set_out_all":                            typeInt,
+	"cmd_other_count":                            typeInt,
+	"cmd_delete_out_all":                         typeInt,
+	"cmd_lease_set_out_all":                      typeInt,
+}
+
+// SampleConfig returns sample configuration message
+func (m *Mcrouter) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns description of Mcrouter plugin
+func (m *Mcrouter) Description() string {
+	return "Read metrics from one or many mcrouter servers"
+}
+
+// Gather reads stats from all configured servers accumulates stats
+func (m *Mcrouter) Gather(acc telegraf.Accumulator) error {
+	ctx := context.Background()
+
+	if m.Timeout.Duration < 1*time.Second {
+		m.Timeout.Duration = defaultTimeout
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, m.Timeout.Duration)
+	defer cancel()
+
+	if len(m.Servers) == 0 {
+		m.Servers = []string{":11211"}
+	}
+
+	for _, serverAddress := range m.Servers {
+		acc.AddError(m.gatherServer(ctx, serverAddress, acc))
+	}
+
+	return nil
+}
+
+func (m *Mcrouter) gatherServer(ctx context.Context, address string, acc telegraf.Accumulator) error {
+	var conn net.Conn
+	var err error
+	var protocol string
+	var dialer net.Dialer
+
+	s := strings.SplitN(address, "://", 2)
```

Comment:
Use `url.Parse` here, I think it will help simplify things and prevent errors.

Reply:
I updated the code to use `url.Parse`

I feel like it makes it a bit stricter on the server connection string format, but I think that's ok.

I make some attempt to tease partial connection strings into a proper format, but ultimately default to `tcp://localhost:11211` if the server connection string does not parse properly.